src/formatting/string.rs

```diff
@@ -225,7 +225,25 @@ fn not_whitespace_except_line_feed(g: &str) -> bool {
 /// character is either a punctuation or a whitespace.
 /// FIXME(issue#3281): We must follow UAX#14 algorithm instead of this.
 fn break_string(max_width: usize, trim_end: bool, line_end: &str, input: &[&str]) -> SnippetState {
-    let break_at = |index /* grapheme at index is included */| {
+    let break_at = |index_in /* grapheme at index is included */| {
+        // Ensure break is not after an escape '\' as it will "escape" the `\` that is added
+        // for concatenating the two parts of the broken line.
+        let index = if (input[index_in] as &str).ne("\\") {
```

Comment:
```suggestion
        let index = if input[index_in] != "\\" {
```

Reply:
OK. Submitted the change (and I have still much to learn about rust, since I was not even aware that `!=` and `ne` are not the same).